library/lua/gui/widgets/text_area/text_area_content.lua

```diff
@@ -0,0 +1,677 @@
+local gui = require('gui')
+local common = require('gui.widgets.common')
+local Widget = require('gui.widgets.widget')
+local WrappedText = require('gui.widgets.text_area.wrapped_text')
+local HistoryStore = require('gui.widgets.text_area.history_store')
+
+local CLIPBOARD_MODE = {LOCAL = 1, LINE = 2}
+local HISTORY_ENTRY = HistoryStore.HISTORY_ENTRY
+
+TextAreaContent = defclass(TextAreaContent, Widget)
+
+TextAreaContent.ATTRS{
+    text = '',
+    text_pen = COLOR_LIGHTCYAN,
+    ignore_keys = {'STRING_A096'},
+    pen_selection = COLOR_CYAN,
+    on_text_change = DEFAULT_NIL,
+    on_cursor_change = DEFAULT_NIL,
+    enable_cursor_blink = true,
+    debug = false,
+    one_line_mode = false,
+    history_size = 10,
+}
+
+function TextAreaContent:init()
+    self.sel_end = nil
+    self.clipboard = nil
+    self.clipboard_mode = CLIPBOARD_MODE.LOCAL
+    self.render_start_line_y = 1
+
+    self.cursor = nil
+
+    self.main_pen = dfhack.pen.parse({
+        fg=self.text_pen,
+        bg=COLOR_RESET,
+        bold=true
+    })
+    self.sel_pen = dfhack.pen.parse({
+        fg=self.text_pen,
+        bg=self.pen_selection,
+        bold=true
+    })
+
+    self.text = self:normalizeText(self.text)
+
+    self.wrapped_text = WrappedText{
+        text=self.text,
+        wrap_width=256
+    }
+
+    self.history = HistoryStore{history_size=self.history_size}
+end
+
+function TextAreaContent:normalizeText(text)
+    if self.one_line_mode then
+        return text:gsub("\r?\n", "")
+    end
+
+    return text
+end
+
+function TextAreaContent:setRenderStartLineY(render_start_line_y)
+    self.render_start_line_y = render_start_line_y
+end
+
+function TextAreaContent:getPreferredFocusState()
+    return true
+end
```

Comment:
it doesn't make sense for both TextArea and TextAreaContent to want focus since TextAreaContent is a child of TextArea. This function should be removed since it is TextArea that really needs the focus.